csrc/python_frontend/fusion_record.h

```diff
@@ -2044,106 +2028,23 @@ struct ScalarRecord : RecordFunctor {
 
   std::pair<serde::RecordData, flatbuffers::Offset<void>> recordData(
       flatbuffers::FlatBufferBuilder& builder) const final {
-    return valueRecordData(builder, value_);
-  };
+    return {
+        serde::RecordData_Scalar,
+        serde::serializeScalar(builder, value_, serde::mapToSerdeDtype(dtype_))
```

Comment:
Nitpick: I think it is slightly cleaner to pass `dtype_` as-is and then convert to `serde dtype` in `serializeScalar`.

Reply:
fixed